src/inverted_index.c

```diff
@@ -141,11 +142,148 @@ ENCODER(encodeDocIdsOnly) {
   return WriteVarint(delta, bw);
 }
 
+/**
```

Comment:
All this needs a bit of documentation, and I'd also split all this logic to a separate C file (probably included directly into inverted_index.c for inlining).